docs/content/docs/concepts/codecs.mdx

```diff
@@ -3,4 +3,2527 @@ title: Codecs
+| [`@solana/codecs-core`](https://www.npmjs.com/package/@solana/codecs-core)                       | Core types and utilities for building codecs.                                      |
+| [`@solana/codecs-numbers`](https://www.npmjs.com/package/@solana/codecs-numbers)                 | Codecs for numbers of various sizes and characteristics.                           |
+| [`@solana/codecs-strings`](https://www.npmjs.com/package/@solana/codecs-strings)                 | Codecs for strings of various encodings and size strategies.                       |
+| [`@solana/codecs-data-structures`](https://www.npmjs.com/package/@solana/codecs-data-structures) | Codecs for a variety of data structures such as objects, enums, arrays, maps, etc. |
+| [`@solana/options`](https://www.npmjs.com/package/@solana/options)                               | Codecs for Rust-list `Options` in JavaScript.                                      |
+
+## What is a Codec?
+
+A Codec is an object that knows how to encode a any type into a `Uint8Array` and how to decode a `Uint8Array` back into that value.
+
+No matter which serialization strategy we use, Codecs abstract away its implementation and offer a simple encode and decode interface. They are also highly composable, allowing us to build complex data structures from simple building blocks.
+
+Here's a quick example that encodes and decodes a simple `Person` type.
+
+```ts twoslash
+import { Codec, addCodecSizePrefix, getUtf8Codec, getU32Codec, getStructCodec, ReadonlyUint8Array } from "@solana/kit";
+// ---cut-before---
+// Use composable codecs to build complex data structures.
+type Person = { name: string; age: number };
+const getPersonCodec = (): Codec<Person> =>
+  getStructCodec([
+    ["name", addCodecSizePrefix(getUtf8Codec(), getU32Codec())],
+    ["age", getU32Codec()],
+  ]);
+
+// Use your own codecs to encode and decode data.
+const personCodec = getPersonCodec();
+const encodedPerson = personCodec.encode({ name: "John", age: 42 });
+const decodedPerson = personCodec.decode(encodedPerson);
+```
+
+## Composing codecs
+
+The easiest way to create your own codecs is to compose the [various codecs](#available-codecs) at your disposal.
+
+For instance, consider the following codecs available:
+
+- `getStructCodec`: Creates a codec for objects with named fields.
+- `getU32Codec`: Creates a codec for unsigned 32-bit integers.
+- `getUtf8Codec`: Creates a codec for UTF-8 strings.
+- `addCodecSizePrefix`: Creates a codec that prefixes the encoded data with its length.
+- `getBooleanCodec`: Creates a codec for booleans using a single byte.
+
+By combining them together we can create a custom codec for the following `Person` type.
+
+```ts twoslash
+import { Codec, addCodecSizePrefix, getUtf8Codec, getU32Codec, getStructCodec, getBooleanCodec } from "@solana/kit";
+
```

Comment:
When do you decide to cut-before and when not to? I'd be inclined to do it an awful lot, unless the intent is for people to be able to straight up copy/paste and run code.

Reply:
Oh I just missed the `cut-before` on that particular example haha.

My logic for including / excluding import statement is: if we're providing a step by step tutorial where people really need the code to run as-is, we include them, otherwise we don't.